packages/binary/messagelayer/messagerequester/messagerequester.go

```diff
@@ -33,17 +33,18 @@ func New(optionalOptions ...Option) *MessageRequester {
 // StartRequest initiates a regular triggering of the StartRequest event until it has been stopped using StopRequest.
 func (requester *MessageRequester) StartRequest(id message.Id) {
 	requester.scheduledRequestsMutex.Lock()
-	defer requester.scheduledRequestsMutex.Unlock()
 
 	// ignore already scheduled requests
 	if _, exists := requester.scheduledRequests[id]; exists {
+		requester.scheduledRequestsMutex.Unlock()
 		return
 	}
 
 	// trigger the event and schedule the next request
 	// make this atomic to be sure that a successive call of StartRequest does not trigger again
```

Comment:
These comments are no longer correct, as now the lock is released after the schedule. They are no longer "atomic".

Reply:
BTW, do you think that it was better before?

Reply:
Ideally,
- the event should not be triggered in the lock,
- the reschedule should happen after the trigger (to account for potentially longer running events,
- should use `defer` for cleaner code and panic safety.

With this PR, it achieves the first point, previously it only achieved the last two :shrug: 
I thought about a solution for all points, but that seemed rather extensive for something that is going to be replaced anyways soon.

Honestly, I was so in the `defer` mindset that I didn't even consider this version :grin: 
I guess it makes sense, the reschedule is not relevant and replacing `defer` for parallel triggers seems reasonable...